modules/apps/frontend-token/frontend-token-definition-impl/src/main/java/com/liferay/frontend/token/definition/internal/FrontendTokenDefinitionManager.java

```diff
@@ -0,0 +1,159 @@
+		_jsonFactory = jsonFactory;
+
+		_themeIdFrontendTokenDefinitionImpls = themeIdFrontendTokenDefinitions;
+		_themeIdFrontendTokenDefinitionImplsDCLSingleton =
+			themeIdFrontendTokenDefinitionsDCLSingleton;
+	}
+
+	public void addFrontendTokenDefinition(
+		long companyId, String externalReferenceCode,
+		String frontendTokenDefinitionJSONString) {
+
+		ResourceBundleLoader resourceBundleLoader =
+			ResourceBundleLoaderUtil.getPortalResourceBundleLoader();
+
+		try {
+			Objects.requireNonNull(frontendTokenDefinitionJSONString);
+			Objects.requireNonNull(externalReferenceCode);
+
+			FrontendTokenDefinitionImpl frontendTokenDefinitionImpl =
+				_newFrontendTokenDefinitionImpl(
+					frontendTokenDefinitionJSONString, resourceBundleLoader,
+					externalReferenceCode);
+
+			Map<String, FrontendTokenDefinition> frontendTokenDefinitionsMap =
+				_getCompanyFrontendTokenDefinitionsMap(companyId);
+
+			frontendTokenDefinitionsMap.put(
+				externalReferenceCode, frontendTokenDefinitionImpl);
+		}
+		catch (JSONException jsonException) {
+			throw new RuntimeException(jsonException);
+		}
+	}
+
+	public FrontendTokenDefinitionImpl addFrontendTokenDefinition(
+		String frontendTokenDefinitionJSONString,
+		ResourceBundleLoader resourceBundleLoader, String themeId) {
+
+		try {
+			Objects.requireNonNull(themeId);
+
+			FrontendTokenDefinitionImpl frontendTokenDefinitionImpl =
+				_newFrontendTokenDefinitionImpl(
+					frontendTokenDefinitionJSONString, resourceBundleLoader,
+					themeId);
+
+			_themeIdFrontendTokenDefinitionImpls.put(
+				themeId, frontendTokenDefinitionImpl);
+
+			return frontendTokenDefinitionImpl;
+		}
+		catch (JSONException jsonException) {
+			throw new RuntimeException(jsonException);
+		}
+	}
+
+	public FrontendTokenDefinition getFrontendTokenDefinition(
+		long companyId, String externalReferenceCode) {
+
+		Map<String, FrontendTokenDefinition> frontendTokenDefinitionsMap =
+			_getCompanyFrontendTokenDefinitionsMap(companyId);
+
+		return frontendTokenDefinitionsMap.get(externalReferenceCode);
+	}
+
+	public Map<String, FrontendTokenDefinitionImpl>
+		getThemeIdFrontendTokenDefinitionImpls() {
+
+		return _themeIdFrontendTokenDefinitionImpls;
+	}
+
+	public DCLSingleton<Map<String, FrontendTokenDefinitionImpl>>
+		getThemeIdFrontendTokenDefinitionImplsDCLSingleton() {
+
+		return _themeIdFrontendTokenDefinitionImplsDCLSingleton;
+	}
+
+	public void removeFrontendTokenDefinition(Long companyId, String key) {
+		if (Objects.nonNull(companyId)) {
+			Map<String, FrontendTokenDefinition> frontendTokenDefinitionsMap =
+				_getCompanyFrontendTokenDefinitionsMap(companyId);
+
+			frontendTokenDefinitionsMap.remove(key);
+		}
+		else {
+			_themeIdFrontendTokenDefinitionImpls.remove(key);
+		}
+	}
+
+	public void removeFrontendTokenDefinition(String key) {
+		removeFrontendTokenDefinition(null, key);
+	}
+
+	private Map<String, FrontendTokenDefinition>
+		_getCompanyFrontendTokenDefinitionsMap(long companyId) {
```

Comment:
`_companyFrontendTokenDefinitionsMap` refereces the entire map, you're actually returning an Entry value of a map. So this should probably be called `_getFrontendTokenDefinintionsMap`. Since you have to pass the companyId, it would be inferred that is the `FrontendTokenDefinitonsMap` for that companyId.

Reply:
fixing it